GentApp/ViewModels/BranchViewModel.cs

```diff
@@ -159,29 +158,31 @@ public class BranchViewModel : ViewModelBase {
 			}
 		}
 
-		public void AddEvent(Event newEvent) {
+		public async void AddEvent(Event newEvent) {
 			Events.Add(newEvent);
-
-			SaveBranchCommand.Execute("Event");
+			await SaveBranch();
 			_navigationService.NavigateTo(nameof(BranchEventsPage));
 		}
 
-		public void EditEvent(string title, string description, DateTime startDate, DateTime endDate) {
+		public async Task EditEvent(string title, string description, DateTime startDate, DateTime endDate) {
 			SelectedEvent.Title = title;
 			SelectedEvent.Description = description;
 			SelectedEvent.StartDate = startDate;
 			SelectedEvent.EndDate = endDate;
+			SelectedEvent.Id = Guid.NewGuid().ToString();
 
-			Events[Events.FindIndex(e => e.Id.Equals(SelectedEvent.Id))] = SelectedEvent;
+			CompanyViewModel.SelectedBranch.Events[Events.FindIndex(e => e.Id.Equals(SelectedEvent.Id))] = SelectedEvent;
 
-			SaveBranchCommand.Execute("Event");
+			await _eventService.Update(SelectedEvent);
```

Comment:
same

Reply:
Keb alles moeten aanpassen zodawe enkele requests gebruiken. Tis redelijk ingewikkeld om een parent object up te daten en dan alle children ook up te daten in 1 keer. Ik dacht dat da vanzelf ging maar blijkbaar niet :/